js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+import Runtime from '../../wasm/publish/rive.js';
+
+// #region LoopEvent
+
+// Loop types. The index of the type is the value that comes from Wasm
+export const loopTypes: string[] = ['oneShot', 'loop', 'pingPong'];
+
+/// Loop events are returned through onloop callbacks
+interface LoopEvent {
+  animation: string;
+  name: string;
+  type: number;
+}
+
+// Creates a new LoopEvent
+export let createLoopEvent = (animation: string, loopValue: number): LoopEvent => {
+  if (loopValue < 0 || loopValue >= loopTypes.length) {
+    throw 'Invalid loop value';
+  }
+  return {
+    animation: animation,
+    type: loopValue,
+    name: loopTypes[loopValue],
+  }
+}
+
+// #endregion
+
+// Maps the playback state to the Wasm enum values
+export const playbackStates = { 'play': 0, 'pause': 1, 'stop': 2 };
+
+// #region layout
+
+// Fit options for the canvas
+export enum Fit {
+  Cover     = 'cover',
```

Comment:
I usually don't recommend to use enum when it's point to a string value. String literal have lighter footprint (it's only a type so it's not complied in the end) and provide a better devX most of the time :
```typescript
type Fit = 'cover' | 'contain' | ...;
```